### PR TITLE
#410 refer to static methods via their class and not via their instance

### PR DESCRIPTION
**Pull request by sjaakd**

### CODE REVIEW

File: processor/src/main/resources/org.mapstruct.ap.model.DefaultMapperReference.ftl [hunk 1]
```diff
@@ -18,4 +18,4 @@
      limitations under the License.
 
 -->
-private final <@includeModel object=type/> ${variableName} = <#if annotatedMapper>Mappers.getMapper( <@includeModel object=type/>.class );<#else>new <@includeModel object=type/>();</#if>
\ No newline at end of file
+<#if used>private final <@includeModel object=type/> ${variableName} = <#if annotatedMapper>Mappers.getMapper( <@includeModel object=type/>.class );<#else>new <@includeModel object=type/>();</#if></#if>
```
Comment (agudian) [inline processor/src/main/resources/org.mapstruct.ap.model.DefaultMapperReference.ftl]: You also need to add that to AnnotationMapperReference.ftl... :)

File: processor/src/test/java/org/mapstruct/ap/test/references/statics/StaticsTest.java [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+ *
+ * @author Sjaak Derksen
+ */
+@IssueKey( "410" )
+@WithClasses( { Beer.class, BeerDto.class, BeerMapper.class, Category.class, CustomMapper.class } )
+@RunWith(AnnotationProcessorTestRunner.class)
+public class StaticsTest {
+
+    @Test
+    public void shouldUseStaticMethod() {
```
Comment (agudian) [inline processor/src/test/java/org/mapstruct/ap/test/references/statics/StaticsTest.java]: Your Test could also assert on the non-existing fields for unused mappers. See ConflictingTypesNamesTest for an example. You could even extend JavaFileAssert with "containsField(...)" or "containsNoField(...)" ... :wink:

Comment (sjaakd) [inline processor/src/test/java/org/mapstruct/ap/test/references/statics/StaticsTest.java]: Did you check the `CustomMapper.class`? It has a private constructor. That serves 2 purposes:
1. Making sure an instance variable cannot be instantiated without compilation errors
2. Keeping check style happy which keeps insisting that utility classes should have private constructors.


> You could even extend JavaFileAssert with "containsField(...)" or "containsNoField(...)" ...

Like that. But don't think its needed in this case.

**Comment by sjaakd** [comment]
Ok.. Fixed the `AnnotationMapperReference`. WDYT, can I merge it?

**Comment by agudian** [comment]
Yes, go ahead! :+1:

File: processor/src/test/java/org/mapstruct/ap/test/references/statics/Beer.java [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+package org.mapstruct.ap.test.references.statics;
+
+/**
+ *
+ * @author Sjaak Derksen
+ */
+public class Beer {
```
Comment (gunnarmorling) [inline processor/src/test/java/org/mapstruct/ap/test/references/statics/Beer.java]: :-)

**Comment by gunnarmorling** [comment]
Nice, one. Thx!

**Pull request closed without merging**